### PR TITLE
Initial cluster sandbox code.

### PR DESCRIPTION
**Pull request by thompsonja**
Also removed unused examples/kubernetes/util/get_forwarded_pool.sh file.

This replaces using vitess-up/down scripts, and after this, I believe all of the individual *-up/down.sh scripts in examples/kubernetes can be removed in favor of using helm (documentation would need to be updated as well).

This PR is part of a couple PRs I'll open, including one for triggering tests using this sandbox via github webhooks, documentation updates, etc.

@michael-berlin

### CODE REVIEW

**Comment by thompsonja** [comment]
I believe I addressed all comments, also did some file reorganization. I'll add Dean to see if he has any other python readability comments.

File: test/cluster/sandbox/sandbox.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    if not self.cluster_type:
+      raise SandboxError('Cannot start cluster, no cluster_type defined')
+    self.cluster.start()
+
+  def stop_cluster(self):
+    if not self.cluster:
+      raise SandboxError('Cannot stop cluster, no cluster_type defined')
+    self.cluster.stop()
+
+  def _execute_sandlets(self, start=True, sandlets=None):
```
Comment (dumbunny) [inline test/cluster/sandbox/sandbox.py]: This method is long enough to need a simple docstring.

File: test/cluster/sandbox/sandbox_utils.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    os.remove(symlink_name)
+  os.symlink(timestamped_name, symlink_name)
+  return open(timestamped_name, 'w')
+
+
+def generate_random_name():
+  adjectives = []
+  animals = []
+  with open('naming/adjectives.txt', 'r') as f:
+    adjectives = [a for a in f.read().split('\n') if a]
```
Comment (dumbunny) [inline test/cluster/sandbox/sandbox_utils.py]: adjectives = [l.strip() for l in f if l.strip()]
...
animals = [l.strip() for l in f if l.strip()]

File: test/cluster/sandbox/sandbox_utils.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  timestamped_name = '%s.%s' % (symlink_name, timestamp)
+  if os.path.islink(symlink_name):
+    os.remove(symlink_name)
+  os.symlink(timestamped_name, symlink_name)
+  return open(timestamped_name, 'w')
+
+
+def generate_random_name():
+  adjectives = []
+  animals = []
```
Comment (dumbunny) [inline test/cluster/sandbox/sandbox_utils.py]: Remove the above 2 lines.

File: test/cluster/sandbox/sandlet.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+class SandletComponent(object):
+  """Entity of a sandlet that encapsulates a process or job."""
+
+  def __init__(self, name, sandbox_name):
+    self.name = name
+    self.sandbox_name = sandbox_name
+    self.dependencies = []
+
+  def start(self):
+    logging.info('Starting component %s', self.name)
```
Comment (dumbunny) [inline test/cluster/sandbox/sandlet.py]: Are these real default behaviors for start and stop? It looks like you always override them, so these should be not implemented.

File: test/cluster/sandbox/wait_for_mysql.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    good_tablets = []
+    for tablet in [t for t in tablets if t not in good_tablets]:
+      _, success = vtctl_sandbox.execute_vtctl_command(
+          ['ExecuteFetchAsDba', tablet, 'show databases'],
+          namespace=options.namespace)
+      if success:
+        good_tablets.append(tablet)
+    logging.info('%d of %d tablets good', len(good_tablets), len(tablets))
+    if len(good_tablets) == len(tablets):
+      logging.info('All tablets ready in %f seconds', time.time() - start_time)
```
Comment (dumbunny) [inline test/cluster/sandbox/wait_for_mysql.py]: You are inconsistent with whether you use punctuation in your logging. Looks ghetto.

**Pull request merged**